### PR TITLE
add smart case and enable by default

### PR DESCRIPTION
**Pull request by ijsnow**
This PR adds "smart case" detection to our query parser. Smart case is now enabled by default.

Closes https://github.com/sourcegraph/sourcegraph/issues/2084.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -14,6 +14,7 @@ All notable changes to Sourcegraph are documented in this file.
 ### Changed
 
 - File match search results now show full repo name if there are results from mirrors on different code hosts (e.g. github.com/sourcegraph/sourcegraph and gitlab.com/sourcegraph/sourcegraph)
+- Search queries now use "smart case" by default. Searches are case insensitive unless you use uppercase letters. To explicitely set the case, you can still use the `case` field (e.g. `case:yes`, `case:no`, `case:smartcase`).
```
Comment (keegancsmith) [inline CHANGELOG.md]: `case:auto` would match other systems that do this behaviour.

File: cmd/frontend/internal/pkg/search/query/searchquery.go [hunk 2]
```diff
@@ -40,7 +42,7 @@ var (
 	conf = types.Config{
 		FieldTypes: map[string]types.FieldType{
 			FieldDefault:   {Literal: types.RegexpType, Quoted: types.StringType},
-			FieldCase:      {Literal: types.BoolType, Quoted: types.BoolType, Singular: true},
+			FieldCase:      {Literal: types.StringType, Quoted: types.StringType, Singular: true},
```
Comment (nicksnyder) [inline cmd/frontend/internal/pkg/search/query/searchquery.go]: Will this break old saved searches? Do we need to do a migration?

File: cmd/frontend/internal/pkg/search/query/searchquery_test.go [hunk 3]
```diff
@@ -10,39 +10,37 @@ import (
...
-		if query.IsCaseSensitive() {
-			t.Error("IsCaseSensitive() == true, want false")
-		}
-	})
+	tests := []struct {
+		name  string
+		input string
+		want  bool
+	}{
+		{"yes", "case:yes", true},
```
Comment (nicksnyder) [inline cmd/frontend/internal/pkg/search/query/searchquery_test.go]: I don't see any test cases for `case:yes` or `case:no`.

File: CHANGELOG.md (hunk 1 above)
Comment (ijsnow) [reply CHANGELOG.md]: `auto` is also shorter

**Pull request merged**